rotkehlchen/chain/ethereum/modules/balancer/balancer.py

```diff
@@ -373,8 +647,386 @@ def _get_address_to_swaps_graph(
+            '$balance': 'BigDecimal!',
+        }
+        param_values = {
+            'limit': GRAPH_QUERY_LIMIT,
+            'offset': 0,
+            'addresses': addresses_lower,
+            'balance': '0',
+        }
+        address_to_pool_balances: DDAddressToPoolBalances = defaultdict(list)
+        while True:
+            try:
+                result = self.graph.query(
+                    querystr=querystr,
+                    param_types=param_types,
+                    param_values=param_values,
+                )
+            except RemoteError as e:
+                self.msg_aggregator.add_error(SUBGRAPH_REMOTE_ERROR_MSG.format(error_msg=str(e)))
+                raise
+
+            try:
+                raw_pool_shares = result['poolShares']
+            except KeyError as e:
+                log.error(
+                    'Failed to deserialize balancer balances',
+                    error='Missing key: poolShares',
+                    result=result,
+                    param_values=param_values,
+                )
+                raise RemoteError('Failed to deserialize balancer balances') from e
+
+            for raw_pool_share in raw_pool_shares:
+                try:
+                    address, balancer_pool = deserialize_pool_share(raw_pool_share)
+                    for pool_token in balancer_pool.tokens:
+                        if isinstance(pool_token.token, EthereumToken):
+                            known_tokens.add(pool_token.token)
+                        elif isinstance(pool_token.token, UnknownEthereumToken):
+                            unknown_tokens.add(pool_token.token)
+                        else:
+                            raise AssertionError(f'Unexpected type: {type(pool_token.token)}')
+                except DeserializationError as e:
+                    log.error(
+                        'Failed to deserialize a balancer pool balance',
+                        error=str(e),
+                        raw_pool_share=raw_pool_share,
+                        param_values=param_values,
+                    )
+                    raise RemoteError('Failed to deserialize balancer balances') from e
+
+                address_to_pool_balances[address].append(balancer_pool)
+
+            if len(raw_pool_shares) < GRAPH_QUERY_LIMIT:
+                break
+
+            param_values = {
+                **param_values,
+                'offset': param_values['offset'] + GRAPH_QUERY_LIMIT,  # type: ignore
+            }
+
+        protocol_balance = ProtocolBalance(
+            address_to_pool_balances=dict(address_to_pool_balances),
+            known_tokens=known_tokens,
+            unknown_tokens=unknown_tokens,
+        )
+        return protocol_balance
+
+    def _get_token_price_at_timestamp_zero_if_error(
+            self,
+            token: Union[EthereumToken, UnknownEthereumToken],
+            timestamp: Timestamp,
+    ) -> Price:
+        usd_price = Price(ZERO)
```

Comment:
This initialization is not needed here as it's already initialized in both `if` and `elif` as also LGTM warns you about: https://lgtm.com/projects/g/rotki/rotki/rev/pr-6953a4d953b922e1f49677c4c17da8e5459715eb